# Board configuration for the environment


board_config:
  #Possible dictionaries:
  # DICT_4X4_50, DICT_4X4_100, DICT_4X4_250, DICT_4X4_1000
  # DICT_5X5_50, DICT_5X5_100, DICT_5X5_250, DICT_5X5_1000
  # DICT_6X6_50, DICT_6X6_100, DICT_6X6_250, DICT_6X6_1000
  # DICT_7X7_50, DICT_7X7_100, DICT_7X7_250, DICT_7X7_1000
  # DICT_ARUCO_ORIGINAL
  dictionary: DICT_5X5_100

  # Sizes should be set to some factor of the output
  #  resolution when generating images
  # Sizes should reflect the real life marker sizes
  #  when detecting
  marker_size: 0.133203125 #values for detection with board size 0.3m in gazebo
  marker_spacing: 0.02578125 #can be calculated by measuring the pixels of each board part and calculating their percentage of the total image. Then the percentage has to be multiplied by the size in Gazebo.
                        #e.g. image has 768x768 px, each marker has 341 px and space between markers is 66 px: marker_size = 341/768 * 0.3, marker_spacing = 66/768 * 0.3
  border_bits: 1 
  output_height: 768  
  output_width: 768

# Set board_id: -1 for these to be incrementally generated.
# Boards should be organized by id (with -1's at tail of list)
# Be aware of dictionary limitations as this will throw
#  errors if you attempt to generate past the
#  dictionary limit.
boards:
  board_0: #has to be first board name
    id: 1 #starting with id 1 for the board on top of Jackal1. Id 0 is always Jackal0, which does not require a board
    rows: 2
    cols: 2
    marker_ids: []
  board_1:
    id: -1
    rows: 2
    cols: 2
    marker_ids: []
  board_2:
    id: -1
    rows: 2
    cols: 2
    marker_ids: []
  board_3:
    id: -1
    rows: 2
    cols: 2
    marker_ids: []
  board_4:
    id: -1
    rows: 2
    cols: 2
    marker_ids: []
  board_5:
    id: -1
    rows: 2
    cols: 2
    marker_ids: []
  board_6:
    id: -1
    rows: 2
    cols: 2
    marker_ids: []
  board_7:
    id: -1
    rows: 2
    cols: 2
    marker_ids: []
  board_8:
    id: -1
    rows: 2
    cols: 2
    marker_ids: []
  board_9:
    id: -1
    rows: 2
    cols: 2
    marker_ids: []
  board_10:
    id: -1
    rows: 2
    cols: 2
    marker_ids: []
  board_11:
    id: -1
    rows: 2
    cols: 2
    marker_ids: []
  board_12:
    id: -1
    rows: 2
    cols: 2
    marker_ids: []
  board_13:
    id: -1
    rows: 2
    cols: 2
    marker_ids: []
  board_14:
    id: -1
    rows: 2
    cols: 2
    marker_ids: []
  board_15:
    id: -1
    rows: 2
    cols: 2
    marker_ids: []
  board_16:
    id: -1
    rows: 2
    cols: 2
    marker_ids: []
  board_17:
    id: -1
    rows: 2
    cols: 2
    marker_ids: []
 